### PR TITLE
Documentation and structure

### PR DESCRIPTION
**Pull request by MansMeg**
Here are the documentation after today's discussion. Would be great with your comments @avehtari and @paul-buerkner . Especially regarding the gold standard definition.

### CODE REVIEW

**Comment by MansMeg** [comment]
@paul-buerkner and @avehtari It would be great to get your comments on the documentation.

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -49,16 +57,11 @@ To see what qualifies as a gold standard, see GOLD_STANDARD_DEFINITION.md.
...
--------------
-
-1. Fork this repository and clone your fork locally
-1. Install pre-commit (see **Installing pre-commit** below)
-1. Make changes and commit them
-1. Run tests (see **Tests**)
-1. Push to your fork
-1. Open a pull request in GitHub
+Below are more details on how to run pre-commits for code cleaning and
+test to test your contribution locally.
```
Comment (paul-buerkner) [inline doc/CONTRIBUTING.md]: 'tests' instead of 'test'?

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -3,7 +3,15 @@ Contributing
...
 
-Below are checklists depending on the what you want to contribute.
+Don't hesitate to do a PR. We use the  PR for discussion on new material!
+
+Pull request workflow
+-------------
+
+1. Fork this repository
+1. Make a contribution
+1. Open a pull request in GitHub (thet test will run on Travis)
```
Comment (paul-buerkner) [inline doc/CONTRIBUTING.md]: on Github?

You mean "the" instead of "thet"?

File: doc/CONTRIBUTING.md [hunk 3]
```diff
@@ -3,7 +3,15 @@ Contributing
 
 We are happy for you to contribute with new posteriors to the datasbase. just create a Pull Request with you new content to the posteriordb, the tests on the new content will be checked.
 
-Below are checklists depending on the what you want to contribute.
+Don't hesitate to do a PR. We use the  PR for discussion on new material!
```
Comment (paul-buerkner) [inline doc/CONTRIBUTING.md]: move this into the preceding paragraph. Explain the abbreviation PR if you use it as some users may not have heard of it. I think the correct term is "make a PR" but I may be wrong.

**Comment by MansMeg** [comment]
Thanks, Paul. Any thoughts on the gold standard definition?

**Comment by paul-buerkner** [comment]
I have looked at the gold standard doc again and have a few comments:

- Link to our Rhat paper so that users know what we mean by "latest version" of Rhat.
- For what purpose do we have an upper bound of the effective samples size? Wouldn't it be sufficient to provide a lower bound?
- I would write "Although other approaches can also be used in special circumstances (e.g., for discrete parameter models) if a clear point can be made that this is necessary."
- In addition to divergent transititions, there should be no draws exceeding the maximum treedepth.

**Comment by avehtari** [comment]
> For what purpose do we have an upper bound of the effective samples size? Wouldn't it be sufficient to provide a lower bound?

We want almost independent draws. If effective sample size is larger than the nominal, then there is dependency which makes further analysis more difficult.

> In addition to divergent transititions, there should be no draws exceeding the maximum treedepth

No. Exceeding max treedepth doesn't invalidate the Markov chain, it just indicates potential performance issues, but even then limiting max treedepth to lower values can gives us improved ESS per log density evaluation.

**Comment by paul-buerkner** [comment]
Ok thanks for the clarifications!

**Comment by MansMeg** [comment]
Great! This is great! I'm currently fixing the gold standards to conform to this, by including both chains when stan_sampling has been used and 

Also. We would like to flag funnel posteriors and multimodal posteriors, and maybe even convex posteriors. But the best would be to just flag this from the posterior samples rather than manually annotating them. My guess is that it should be possible to estimate funnels based on 10 000 samples? The same with multimodal posteriors? What do you say?

**Comment by paul-buerkner** [comment]
I we have no divergent transitions, that chances are we have captured the funnel if the chains ever came remotely close the the funnel.

For multimodality there is not quanrantee that we capture all the modes. It might as well be that we simply missed some modes. Ideally, we should work with multimodal distribution where we understand how many modes we have and as such can check if we found all of them. But in a lot of cases multimodality indicates some way of model misspecification (for instance if we forgot to identify mixture compoenents).

**Comment by MansMeg** [comment]
Alright, changes added. I'll merge later today if no one has any comments.

**Comment by eerolinna** [comment]
> We would like to flag funnel posteriors and multimodal posteriors, and maybe even convex posteriors. But the best would be to just flag this from the posterior samples rather than manually annotating them. My guess is that it should be possible to estimate funnels based on 10 000 samples? The same with multimodal posteriors? What do you say?

Could we use the multimodality/funnel information to help validate the gold standards? So for example we know that some posterior should have 3 distinct modes. However the estimate says that there are only two, this might mean that the samples are not a valid gold standard (or it can mean that the estimate is wrong). Lets say we have 100 manual annotations, if we can catch 1 bad gold standard with them it would probably be worth the effort.

Of course this requires that we actually know the true number/location of modes etc so I don't know how useful this would be in practice.

**Comment by MansMeg** [comment]
No, we do not need this. I spoke with Aki and he thought it would be good to have this functionality. Although - only as suggestions. We want humans to add all keywords manually.

**Pull request merged**